ui/src/components/CoreConcepts/Counter.js

```diff
@@ -0,0 +1,47 @@
+import React, {useState} from 'react';
+import { Container, Row, Col } from 'react-bootstrap';
+import CounterDisplay from './CounterDisplay';
+import IncrementButton from './IncrementButton';
+import DecrementButton from './DecrementButton';
+import ResetButton from './ResetButton';
+
+function Counter () {
+    const [count, setCount] = useState(0);
+
+    function increment () {
+        setCount(count + 1);
+    }
+
+    function decrement () {
+        setCount(count - 1);
+    }
+
+    function reset () {
+        setCount(0);
+    }
+
+    return (
+           <Container className='counter-container'>
+              <Row>
+                  <Col>
+                       <CounterDisplay count={count} />
+                 </Col>
+             </Row>
+             <Row>
+                  <Col>
+                       <IncrementButton increment={increment}/>
```

Comment:
Again about naming conventions. Props that receive event handlers should be called `onSomething` because it mimics the convention of the regular HTML (`onclick` etc.). Here you're not even supposed to mention increment in the name of the prop, it should just be `onClick.` The `IncrementButton` shouldn't know what the handler does, it should only know about _on which event_ to call the handler. Internally it calls the handler on click , so the prop should be `onClick`